src/index.ts

```diff
@@ -0,0 +1,176 @@
+import { SoftwareKeyProvider } from 'jolocom-lib'
+import { BaseMetadata } from '@jolocom/protocol-ts'
+import {
+  ICredentialRequest,
+  ICredentialRequestAttrs,
+  CredentialOffer,
+  CredentialOfferRequestAttrs,
+  ICredentialsReceiveAttrs,
+} from 'jolocom-lib/js/interactionTokens/interactionTokens.types'
+import { ISignedCredCreationArgs } from 'jolocom-lib/js/credentials/signedCredential/types'
+import { SDKError, ErrorCode } from './errors'
+export { SDKError, ErrorCode }
+
+export {
+  ICredentialRequest as CredentialRequirements,
+  ICredentialRequestAttrs as CredentialRequest,
+  CredentialOffer,
+  CredentialOfferRequestAttrs,
+  ICredentialsReceiveAttrs as CredentialPayload,
+  BaseMetadata as CredentialDefinition,
+  ISignedCredCreationArgs as CredentialData,
+}
+import { IStorage, IPasswordStore } from './storage'
+export { NaivePasswordStore } from './storage'
+export { JolocomLib } from 'jolocom-lib'
+export { JSONWebToken } from 'jolocom-lib/js/interactionTokens/JSONWebToken'
+import { InternalDb } from '@jolocom/local-resolver-registrar/js/db'
+import { DidMethodKeeper } from './didMethodKeeper'
+import { LocalDidMethod } from 'jolocom-lib/js/didMethods/local'
+import { IResolver } from 'jolocom-lib/js/didMethods/types'
+import { Identity } from 'jolocom-lib/js/identity/identity'
+import { Agent } from './agent'
+export { Agent } from './agent'
+
+export interface IJolocomSDKConfig {
+  storage: IStorage
+  eventDB?: InternalDb
+}
+
+export interface IJolocomSDKInitOptions {
+  did?: string
+  mnemonic?: string
+  auto?: boolean
+  pass?: string
+  passOrStore?: string | IPasswordStore
+}
+
+export interface JolocomPlugin {
+  register(sdk: JolocomSDK): Promise<void>
+}
+
+export class JolocomSDK {
+  public didMethods = new DidMethodKeeper()
+  public storage: IStorage
+  public resolver: IResolver
+
+  constructor(conf: IJolocomSDKConfig) {
+    this.storage = conf.storage
+    const localDidMethod = new LocalDidMethod(
+      conf.eventDB || this.storage.eventDB,
+    )
+    this.didMethods.register('jun', localDidMethod)
+    // FIXME the prefix bit is required just to match IResolver
+    // but does anything need it at that level?
```

Comment:
this is so that IResolvers for a given method can be found when aggregated together. if this.resolver is used in that context there may be strange behaviour